ppmclibs/tinycv_impl.cc

```diff
@@ -617,6 +617,28 @@ VNCInfo* image_vncinfo(bool do_endian_conversion, bool true_colour,
         blue_shift);
 }
 
+/*
+ * Fill given rectangle with pixel value read from 'data' and interpreted
+ * according to 'info'. Number of bytes read from 'data' depends on 'info'.
+ */
+void image_fill_pixel(Image* a, const unsigned char *data, VNCInfo* info,
+    long x, long y, long width, long height)
+{
+    size_t offset = 0;
+    Vec3b pixel = info->read_pixel(data, offset);
+
+    // avoid an exception
+    if (x < 0 || y < 0 || y + height > a->img.rows || x + width > a->img.cols) {
+        std::cerr << "ERROR - fill_pixel: out of range\n"
+                  << std::endl;
+        return;
+    }
+
+    for (int i = y; i < y + height; i++)
+        for (int j = x; j < x + width; j++)
```

Comment:
Maybe make the casting explicit:
```suggestion
    for (auto i = static_cast<int>(y); i < y + height; i++)
        for (auto j = static_cast<int>(x); j < x + width; j++)
```